### PR TITLE
Add additional math operations

### PR DESCRIPTION
**Pull request by ayselafsar**
- Get intersection of two 2d lines
- Get intersection of two rectangles
- Add sign operation in math module

### CODE REVIEW

File: src/lineSegment.js [hunk 1]
```diff
@@ -43,10 +43,95 @@ var cornerstoneMath = (function (cornerstoneMath) {
...
+         * on same side of second line segment, the line segments do
+         * not intersect.
+         */
+
+        if (r1 !== 0 &&
+            r2 !== 0 &&
+            cornerstoneMath.sign(r1) === cornerstoneMath.sign(r2)) {
+            intersectionPoint.x = 0;
+            intersectionPoint.y = 0;
+            intersectionPoint.intersected = false;
```
Comment (swederik) [inline src/lineSegment.js]: Can you remove the 'intersected' attribute and just return undefined if there is no intersection?

This function should just return a point object with x and y coordinates, or undefined if no intersection exists.

File: src/rect.js [hunk 2]
```diff
@@ -78,10 +78,110 @@ var cornerstoneMath = (function (cornerstoneMath) {
...
+            if (rect2.height >= 0)
+                intersectTopBottom = !((rect1Points.top <= rect2Points.top) || (rect2Points.bottom <= rect1Points.bottom));
+            else
+                intersectTopBottom = !((rect1Points.top <= rect2Points.bottom) || (rect2Points.top <= rect1Points.bottom));
+        }
+
+        return intersectLeftRight && intersectTopBottom;
+    }
+
+    // Returns intersection points of two rectangles
```
Comment (swederik) [inline src/rect.js]: Could you provide some further documentation?
- Does this only work for non-rotated rectangles?
- It appears that the function returns coordinates as left, right, top, and bottom. Could it be expressed as two points instead?

**Comment by swederik** [comment]
Hi Aysel,

I've made some comments inline which you can see above. Let me know if you need any clarification.

Thanks for the contribution!

File: src/rect.js (hunk 2 above)
Comment (ayselafsar) [inline src/rect.js]: Yes, It only works for non-rotated rectangles.
It is updated to use two points.

File: src/lineSegment.js (hunk 1 above)
Comment (ayselafsar) [inline src/lineSegment.js]: This is done.

**Pull request merged**